### PR TITLE
NCGenerics: more test fixes (v2)

### PR DESCRIPTION
**Pull request by kavon**
base branch is https://github.com/apple/swift/pull/71371

### CODE REVIEW

**Comment by kavon** [comment]
@swift-ci smoke test

File: lib/Sema/TypeCheckDecl.cpp [hunk 1]
```diff
@@ -928,9 +928,8 @@ InvertibleAnnotationRequest::evaluate(Evaluator &evaluator,
   switch (TARGET) {
   case InvertibleProtocolKind::Copyable:
     // Handle the legacy '@_moveOnly' attribute
-    if (auto attr = decl->getAttrs().getAttribute<MoveOnlyAttr>()) {
-      assert((isa<StructDecl, EnumDecl, ClassDecl>(decl)));
-
+    if (auto attr = decl->getAttrs().getAttribute<MoveOnlyAttr>();
+        attr && isa<StructDecl, EnumDecl, ClassDecl>(decl)) {
```
Comment (slavapestov) [inline lib/Sema/TypeCheckDecl.cpp:932]: This will be clearer with a nested `if` I think.

**Comment by kavon** [comment]
@swift-ci smoke test

**Comment by kavon** [comment]
@swift-ci smoke test

**Comment by kavon** [comment]
@swift-ci smoke test

**Comment by kavon** [comment]
preset=ncgenerics,smoketest=macosx
@swift-ci Please test with preset macOS Platform

**Comment by kavon** [comment]
@swift-ci smoke test

**Pull request merged**